usr/src/uts/common/fs/zfs/metaslab.c

```diff
@@ -2652,6 +2730,7 @@ metaslab_block_alloc(metaslab_t *msp, uint64_t size, uint64_t txg)
 		VERIFY0(P2PHASE(size, 1ULL << vd->vdev_ashift));
 		VERIFY3U(range_tree_space(rt) - size, <=, msp->ms_size);
 		range_tree_remove(rt, start, size);
+		metaslab_trim_remove(msp, start, size);
```

Comment:
I'll add the check into metaslab_trim_remove - that covers both sites where this is used.